_posts/2018-06-25-refactoring-with-design-patterns.markdown

```diff
@@ -0,0 +1,192 @@
+---
+layout: post
+title: "Refactoring: clean your ruby code with design patterns"
+date: 2018-06-25 12:00:00
+categories: ["code-refactor"]
+author: "cleiviane"
+---
+
+Code refactoring can be defined as <strong>*“the process of introducing small and incremental changes to leave the code in a better state than it was.”*</strong>. When refactoring your code you have to consider two things: none of new functionality should be added and the external behavior should not be affected.
+
+One of the Ruby on Rails developer challenges is to keep your code clean, simple and easy to maintain and that is why we are always refactoring our code.
+
+There are several techniques that a developer can follow to improve his code by code refactoring, such as extract method, move method, move field, switch statements, etc. If you are not familiarized with them, please visit the [Refactoring Guru site](https://refactoring.guru/).
+
+<!--more-->
```

Comment:
Maybe you can add a phrase before the collapse here so that they are tempted to stay on the blog instead of going to the Refactoring Guru site. Something like `In this post we'll try to go over some of these techniques and how you can apply them to your Ruby code.`